haystack/schema.py

```diff
@@ -540,11 +546,18 @@ def __init__(self,
         self.document_contents = [l.document.content for l in self.labels if not l.no_answer]
 
     def _aggregate_labels(self, key, must_be_single_value=True) -> List[Any]:
-        unique_values = set([getattr(l, key) for l in self.labels])
+        if key == "filters":
```

Comment:
Would be nicer if that would work for all dict fields (like meta). 
```python
if any(isinstance(getattr(l, key), dict) for l in self.labels))
```
should do it.